reframe/frontend/cli.py

```diff
@@ -414,20 +414,30 @@ def main():
             raise ReframeError from e
 
         # Filter checks by name
-        checks_matched = filter(filters.have_not_name(options.exclude_names),
-                                checks_found)
+        checks_matched = checks_found
+        if options.exclude_names:
+            for name in options.exclude_names:
+                checks_matched = filter(filters.have_not_name(name),
+                                        checks_matched)
 
         if options.names:
-            checks_matched = filter(filters.have_name(options.names),
-                                    checks_matched)
+            if len(options.names) > 1:
```

Comment:
Done.

Reply:
I think I need help concerning the wildcard support issue: do you want something like a check on the  valid_prog_environs variable instantiation so that when a test provides as supported programming environment a wildcard that is not "*" an exception is raised?